CMakeLists.txt

```
@@ -21,43 +21,73 @@ if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.12")
     cmake_policy(SET CMP0074 NEW)
 endif()
 
-find_package(ZLIB REQUIRED)
-include_directories(${ZLIB_INCLUDE_DIRS})
-
-set(spng_SOURCES spng/spng.c)
-
 if(NOT CMAKE_HOST_WIN32)
-    set(spng_LIBS -lm ${ZLIB_LIBRARIES})
+    set(MATH_LIBRARY "-lm")
 else()
-    set(spng_LIBS ${ZLIB_LIBRARIES})
+    set(MATH_LIBRARY "")
 endif()
 
 if(NOT ENABLE_OPT)
     add_definitions( -DSPNG_DISABLE_OPT=1 )
 endif()
 
+set(STATIC_SHARED_TARGETS "")
+
+set(spng_SOURCES spng/spng.c)
+
 if(SPNG_SHARED)
     add_library(spng SHARED ${spng_SOURCES})
-    target_include_directories(spng PUBLIC ${PROJECT_SOURCE_DIR}/spng)
-    target_link_libraries(spng ${spng_LIBS})
-    install(TARGETS spng DESTINATION lib)
+    list(APPEND STATIC_SHARED_TARGETS spng)
 
     if(BUILD_EXAMPLES)
         add_executable(example examples/example.c)
-        target_include_directories(example PRIVATE ${PROJECT_SOURCE_DIR}/spng)
-        target_link_libraries(example spng ${spng_LIBS})
+        target_link_libraries(example PRIVATE spng)
     endif()
 endif()
 
 if(SPNG_STATIC)
     add_library(spng_static STATIC ${spng_SOURCES})
-    target_include_directories(spng PUBLIC ${PROJECT_SOURCE_DIR}/spng)
     target_compile_definitions(spng_static PUBLIC SPNG_STATIC)
-    install(TARGETS spng_static DESTINATION lib)
+    list(APPEND STATIC_SHARED_TARGETS spng_static)
 endif()
 
+find_package(ZLIB REQUIRED)
+foreach(STATIC_SHARED_TARGET ${STATIC_SHARED_TARGETS})
+    target_include_directories(${STATIC_SHARED_TARGET} PUBLIC
+	  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/spng>
+	  $<INSTALL_INTERFACE:include/>
```

Please use the variables from the `GNUInstallDirs` module:

```suggestion
	  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/>
```